rust_dev_preview/setting-timeouts/src/main.rs

```diff
@@ -35,22 +41,28 @@ async fn main() -> Result<(), aws_sdk_s3::Error> {
     // Here we create an object that holds timeout-related configuration. We'll have to pass this
     // config into two places. This is because different timeouts are handled at different
     // "levels" of the AWS SDK Client networking stack. We'll note which timeouts are getting
-    // set each time we pass in this cofiguration.
-    let timeout_config = TimeoutConfig::new()
-        // This timeout acts at the "Request to a service" level. When the SDK makes a request to a
-        // service, that "request" may actually comprise of several HTTP requests in order to retry
-        // failures that are likely spurious or to refresh credentials.
-        .with_api_call_timeout(Some(Duration::from_secs(2)))
-        // This timeout acts at the "HTTP request" level and will set a separate timeout for each
-        // HTTP request made as part of a "service request"
-        .with_api_call_attempt_timeout(Some(Duration::from_secs(2)))
-        // A limit on the amount of time an application takes to attempt to read the first byte over
-        // an established, open connection after write request.
-        // Also known as the "time to first byte" timeout
-        .with_read_timeout(Some(Duration::from_secs(2)))
-        // A limit on the amount of time after making an initial connect attempt on a socket to
-        // complete the connect-handshake
-        .with_connect_timeout(Some(Duration::from_secs(2)));
+    // set each time we pass in this configuration.
+    let timeout_config = timeout::Config::new()
+        .with_api_timeouts(
+            timeout::Api::new()
+                // This timeout acts at the "Request to a service" level. When the SDK makes a request to a
+                // service, that "request" may actually comprise of several HTTP requests in order to retry
+                // failures that are likely spurious or to refresh credentials.
```

Comment:
// This timeout acts at the "Request to a service" level. When the SDK makes a request to a
                // service, that "request" can contain several HTTP requests. This way, you can retry
                // failures that are likely spurious, or refresh credentials.